system-addon/content-src/components/Sections/Sections.jsx

```diff
@@ -19,7 +20,14 @@ function getFormattedMessage(message) {
 export class Section extends React.PureComponent {
   _dispatchImpressionStats() {
     const {props} = this;
-    const maxCards = 3 * props.maxRows;
+    let cardsPerRow = CARDS_PER_ROW_DEFAULT;
+    if (props.compactCards && global.matchMedia(`(min-width: 1072px)`).matches) {
+      // If the section has compact cards and the viewport is wide enough, we show
+      // 4 columns instead of 3.
+      // $break-point-widest = 1072px (from _variables.scss)
+      cardsPerRow = CARDS_PER_ROW_COMPACT_WIDE;
```

Comment:
Something about this seems a bit confusing with some implied behavior. I'm not sure if something like `EXTRA_COMPACT_CARDS = 1` as maybe it's clearer that non-wide with compact would just use the default. I guess we'll leave this as is for now.

Reply:
Yeah. it's icky. I need to play with CSS grids a little more. What happens if you specify a 3x3 grid but have 8 items? Do the last 2 get hidden automagically?